src/Sarif.WorkItemFiling/WorkItemFiler.cs

```diff
@@ -67,8 +74,7 @@ public async Task<IEnumerable<ResultGroup>> FileWorkItems(string logFilePath)
             {
                 if (sarifLog.Runs[i]?.Results?.Count > 0)
                 {
-                    // TODO: Extract this filtering logic into some sort of "filtering strategy" object.
-                    IList<Result> filteredResults = FilterResults(sarifLog.Runs[i].Results);
+                    IList<Result> filteredResults = _filteringStrategy.FilterResults(sarifLog.Runs[i].Results);
 
                     // TODO: Consider whether to await each run in turn, or to do them in parallel.
```

Comment:
>or to do them in pa [](start = 73, length = 19)

In general, .NET's web request API is throttled to two connections to the same endpoint. It is a neat trick to try to subvert this and increase parallelism. Since a single run is our happy path, suggest leaving this alone for now.

Reply:
Oh! That explains why, in one of the code bases I looked at (I forget which one), they processed the work items in batches of two. Naturally, there was no comment explaining that ;-)

Ok, I will remove this comment and we'll leave it alone for now.

---
In reply to: [306116599](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306116599) [](ancestors = 306116599)